### PR TITLE
CAMEL-15772: camel-pulsar readCompacted support

### PR DESCRIPTION
**Pull request by ludovic-boutros**
- [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
- [x] Each commit in the pull request should have a meaningful subject line and body.
- [x] If you're unsure, you can format the pull request title like `[CAMEL-XXX] Fixes bug in camel-file component`, where you replace `CAMEL-XXX` with the appropriate JIRA issue.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Run `mvn clean install -Psourcecheck` in your module with source check enabled to make sure basic checks pass and there are no checkstyle violations. A more thorough check will be performed on your pull request automatically.
Below are the contribution guidelines:
https://github.com/apache/camel/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarConfiguration.java [hunk 1]
```diff
@@ -63,6 +63,8 @@
     private long ackGroupTimeMillis = 100;
     @UriParam(label = "consumer", defaultValue = "LATEST")
     private SubscriptionInitialPosition subscriptionInitialPosition = LATEST;
+    @UriParam(label = "consumer", defaultValue = "false")
+    private boolean readCompacted = false;
```
Comment (oscerd) [inline components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarConfiguration.java:67]: The defaultValue at field level is not needed, it is false by default.

**Review by oscerd — CHANGES_REQUESTED** [review]
Just a little thing, looks good to me and thanks a lot for this.

**Review by davsclaus — APPROVED** [review]
Yeah there is only same comment as Andrea. Also you can check the code via mvn compile -P sourcecheck in the camel-pulsar dir

**Comment by ludovic-boutros** [comment]
Should be ok now :) thx !

File: components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarConfiguration.java (hunk 1 above)
Comment (ludovic-boutros) [reply components/camel-pulsar/src/main/java/org/apache/camel/component/pulsar/PulsarConfiguration.java]: done. :)

**Review by oscerd — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

**Comment by ludovic-boutros** [comment]
I need to fully rebuild the project in order to commit this file :
catalog/camel-catalog/src/generated/resources/org/apache/camel/catalog/components/pulsar.json

**Comment by ludovic-boutros** [comment]
It's good now, all checks have passed \o/ 👍

**Pull request merged**